src/meanshift.jl

```diff
@@ -0,0 +1,96 @@
+"""
+```
+segments                = meanshift(img, spatial_radius, range_radius, [iter], [eps])
+``` 
+Segments the image using meanshift clustering.
+    
+Parameters:  
+-    img                            = input grayscale image
+-    spatial_radius/range_radius    = bandwidth parameters of truncated normal kernel. Controlling the size of the kernel determines the resolution of the mode detection.
+-    iter/eps                       = stopping criterion for meanshift procedure. The algorithm stops after iter iterations or if kernel center moves less than eps distance in an update step, whichever comes first.
+"""
+
+function meanshift{CT}(img::Array{CT, 2}, spatial_radius::Real, range_radius::Real; iter::Int = 50, eps::Real = 0.01)
+
+    rows, cols = size(img)
+    rowbins = Int(floor(rows/spatial_radius))
```

Comment:
It would be simple to extend the algorithm to N-dimensional images. Performance-wise I don't think the algorithm(current implementation) scales well to higher dimension.